lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1650,9 +1666,25 @@ def cmd_sessions(*args)
         end
       end
     when 'killall'
-      print_status("Killing all sessions...")
-      framework.sessions.each_sorted.reverse_each do |s|
-        session = framework.sessions.get(s)
+      if search_term
+        matching_sessions = get_matching_sessions(search_term)
+        if matching_sessions.empty?
+          print_status("No matching sessions.") 
+          return
+        end
```

Comment:
You have this block of code copy pasta'd in a few places, I would probably consider doing the search earlier in the code path once and then any subsequent actions can use the results